### PR TITLE
:sparkles: Update to latest cluster-api version

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:
- Updates cluster-api dependencies to v0.3.3
- Replaces custom isPaused function with sigs.k8s.io/cluster-api/util/HasPausedAnnotation

### CODE REVIEW

File: controllers/awscluster_controller.go [hunk 1]
```diff
@@ -73,13 +73,13 @@ func (r *AWSClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reter
...
-		log.Info("AWSCluster or linked Cluster is marked as paused. Won't reconcile")
+	if cluster == nil {
+		log.Info("Cluster Controller has not yet set OwnerRef")
 		return reconcile.Result{}, nil
 	}
 
-	if cluster == nil {
-		log.Info("Cluster Controller has not yet set OwnerRef")
+	if util.IsPaused(cluster, awsCluster) {
+		log.Info("AWSCluster or linked Cluster is marked as paused. Won't reconcile")
```
Comment (detiber) [inline controllers/awscluster_controller.go]: Swapped these to ensure that we aren't passing a nil cluster in to util.IsPaused

File: controllers/helpers.go [hunk 2]
```diff
@@ -48,20 +47,10 @@ func pausedPredicates(logger logr.Logger) predicate.Funcs {
 func processIfUnpaused(logger logr.Logger, obj runtime.Object, meta metav1.Object) bool {
 	kind := strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind)
 	log := logger.WithValues("namespace", meta.GetNamespace(), kind, meta.GetName())
-	if isPaused(nil, meta) {
+	if clusterutil.HasPausedAnnotation(meta) {
```
Comment (randomvariable) [inline controllers/helpers.go]: shouldn't this be clusterutil.IsPaused(... ?

Comment (randomvariable) [reply controllers/helpers.go]: Ah, nevermind. It's not for the cluster object.

Comment (detiber) [reply controllers/helpers.go]: `clusterutil.IsPaused` does not handle a nil cluster, so this basically just checks the paused annotation which is what using a nil cluster with the local `isPaused` would fall back to.

**Comment by ncdc** [comment]
/lgtm

/hold
waiting on e2e results

**Pull request merged**